### PR TITLE
+doc #20794 explain Props edge cases.

### LINKED ISSUE #20794: Creation of actor with default value fail when using `system.actorOf(Props[MyActor])`
Given the following actor class:
```
class MyActor(val arg: Int = 5) extends Actor {
  ...
}
```
Creating an instance of MyActor using either:
```
system.actorOf(Props[MyActor])
system.actorOf(Props(classOf[MyActor]))
```
Throws an 

> IllegalArgumentException: no matching constructor found on MyActor for arguments []

Which is unexpected since the class MyActor should have an empty constructor.

### PR DESCRIPTION
**Pull request by Lasering**
Fixes #20794

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

File: akka-docs/rst/scala/actors.rst [hunk 1]
```diff
@@ -108,6 +108,31 @@ reference needs to be passed as the first argument).
...
+    val valueClassProp = Props(classOf[ValueActor], new DurationInt(5)) // Unsupported
+
+* An actor with default constructor values.
+
+.. includecode:: code/docs/actor/ActorDocSpec.scala#props-edge-cases-default-values
+
+.. code-block:: scala
+    val defaultValueProp1 = Props(classOf[DefaultValueActor], 2.0) // Unsupported
+    val defaultValueProp2 = Props[DefaultValueActor2] // Unsupported
+    val defaultValueProp3 = Props(classOf[DefaultValueActor2]) // Unsupported
```
Comment (ktoso) [inline akka-docs/rst/scala/actors.rst]: please put those in `ActorDocSpec` too, you can make them a `compileOnlySpec` (see in other tests how to)

File: akka-docs/rst/scala/actors.rst [hunk 2]
```diff
@@ -108,6 +108,31 @@ reference needs to be passed as the first argument).
...
+
+In both cases an :class:`IllegalArgumentException` will be thrown stating
+no matching constructor could be found.
+
+The next topic explains a possible way to work around these edge cases.
```
Comment (ktoso) [inline akka-docs/rst/scala/actors.rst]: "The next section explains the recommended ways to create Actor props in a way, which simultaneously safe-guard against these edge cases."?

File: akka-docs/rst/scala/actors.rst [hunk 3]
```diff
@@ -108,6 +108,31 @@ reference needs to be passed as the first argument).
...
+Edge cases
+^^^^^^^^^^
+There are two edge cases in actor creation with :class:`Props`:
+
+* An actor with :class:`AnyVal` arguments.
+
+.. includecode:: code/docs/actor/ActorDocSpec.scala#props-edge-cases-value-class
+
+.. code-block:: scala
+    val valueClassProp = Props(classOf[ValueActor], new DurationInt(5)) // Unsupported
```
Comment (ktoso) [inline akka-docs/rst/scala/actors.rst]: please put in code anyway

File: akka-docs/rst/scala/code/docs/actor/ActorDocSpec.scala [hunk 4]
```diff
@@ -320,6 +320,30 @@ class ActorDocSpec extends AkkaSpec("""
     // encourages to close over enclosing class
     val props7 = Props(new MyActor)
     //#creating-props-deprecated
+
+    //#props-edge-cases-value-class
+    // DurationInt is a Value Class, that is, it extends AnyVal
+    import scala.concurrent.duration.DurationInt
+    class ValueActor(duration: DurationInt) extends Actor {
```
Comment (ktoso) [inline akka-docs/rst/scala/code/docs/actor/ActorDocSpec.scala]: please show the value class instead, like so:

```
case class MyValueClass(v: Int) extends AnyVal
```

**Comment by ktoso** [comment]
Looks good, thanks! Some minor touchups :)

**Comment by drewhk** [comment]
LGTM, thanks for adding these

**Comment by ktoso** [comment]
LGTM, thanks!

**Comment by patriknw** [comment]
thanks @Lasering

**Pull request merged**